src/Couchbase.Lite.Shared/API/Document/Document.cs

```diff
@@ -403,6 +404,10 @@ public void Dispose()
         /// <inheritdoc />
         public string ToJSON()
         {
+            if(IsMutable) {
+                return JsonConvert.SerializeObject(_dict);
```

Comment:
How does this handle Blob objects?

Reply:
I have made modification to the Blob class to make it serializable to Dictionary<string, object> json string represents Blob and Dictionary<string, object> represents Blob deserializable.